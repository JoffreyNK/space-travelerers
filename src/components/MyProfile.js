import React from 'react';
import RocketsProfile from './my-profile/rockets-profile';
import MissionsProfile from './my-profile/missions-profile';
import DragonsProfile from './my-profile/dragons-profile';
import '../index.css';

const MyProfile = () => (
  <div className="pre-pf-container">
    <hr />
    <div className="all-profiles-container">
      <div className="cp1">
        <RocketsProfile />
      </div>
      <div className="cp2">
        <MissionsProfile />
      </div>
      <div className="cp3">
        <DragonsProfile />
      </div>
    </div>
  </div>
);

export default MyProfile;
