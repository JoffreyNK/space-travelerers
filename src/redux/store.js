import { configureStore, combineReducers } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsReducer';

const reducer = combineReducers(
  {
    missions: missionsSlice.reducer,
  },
);

const store = configureStore({ reducer });

export default store;
