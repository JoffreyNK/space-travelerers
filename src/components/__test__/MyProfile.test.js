// MyProfile.test.js

import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MyProfile from '../MyProfile';
import store from '../../redux/store';

describe(' Rockets Component Rendering ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
