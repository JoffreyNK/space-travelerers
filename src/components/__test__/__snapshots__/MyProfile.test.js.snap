// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Rockets Component Rendering  renders correctly 1`] = `
<div
  className="all-profiles-container"
>
  <div
    className="cp1"
  >
    <div
      className="pf-container"
    >
      <h2
        className="pf-title"
      >
        My Rockets
      </h2>
      <table
        className="pf-table"
      >
        <tbody
          className="pf-item-list"
        />
      </table>
    </div>
  </div>
  <div
    className="cp2"
  >
    <div
      className="pf-container"
    >
      <h2
        className="pf-title"
      >
        My Missions
      </h2>
      <table
        className="pf-table"
      >
        <tbody
          className="pf-item-list"
        />
      </table>
    </div>
  </div>
  <div
    className="cp3"
  >
    <div
      className="pf-container"
    >
      <h2
        className="pf-title"
      >
        My Dragons
      </h2>
      <table
        className="pf-table"
      >
        <tbody
          className="pf-item-list"
        />
      </table>
    </div>
  </div>
</div>
`;
